### PR TITLE
SecretStore: documentation revamp

### PR DESCRIPTION
**Pull request by svyatonik**
There are a lot of problems with current Secret Store documentation - I'm willing to make a new one, which is more structurized, is up-to-date and have answers to common questions.

### CODE REVIEW

File: JSONRPC.md [hunk 1]
```diff
@@ -50,3 +50,4 @@ As documented in the options, available under `parity --help` not all API's are
 - [signer](JSONRPC-signer-module.md)
 - [trace](JSONRPC-trace-module.md)
 - [shh](JSONRPC-shh-Module.md)
+- [secretstore](Secret-Store-RPC.md)
```
Comment (5chdn) [inline JSONRPC.md]: can you rename the file to `JSONRPC-secretstore-Module.md` ?

File: _includes/_Sidebar.md [hunk 2]
```diff
@@ -57,6 +57,7 @@
   - [signer](JSONRPC-signer-module)
   - [trace](JSONRPC-trace-module)
   - [shh](JSONRPC-shh-Module)
+  - [secretstore](Secret-Store-RPC.md)
```
Comment (5chdn) [inline _includes/_Sidebar.md]: ~~Not sure if the generated HTML wiki will work with `.md` extension~~

Edit, just tested, this works :+1:

**Comment by 5chdn** [comment]
I did not try to run through the full docs, but it looks good, let's just keep the file names consistent.

**Comment by svyatonik** [comment]
@5chdn Thanks! Renamed

**Pull request merged**